src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -14,10 +17,13 @@
 namespace Microsoft.DotNet.Build.Tasks
 {
     /*
-     * This task updates the package source mappings in the NuGet.Config.
-     * If package source mappings are used, source-build packages sources will be added with the cumulative package patterns
-     * for all of the existing package sources. When building offline, the existing package source mappings will be removed;
-     * otherwise they will be preserved after the source-build sources.
+     * This task updates the package source mappings in the NuGet.Config using the following logic:
+     * Add all packages from current source-build sources, i.e. source-built-*, reference-packages.
+     * For previously source-built sources, add only the packages that do not exist in any of the current source-built sources.
+     * Also add PSB packages if that package version does not exist in current package sources.
+     * In offline build, remove all existing package source mappings for online sources.
```

Comment:
Isn't this done by RemoveInternetSourcesFromNuGetConfig?

Reply:
Yes, correct. This task also contains the code for removing existing package mappings, which wasn't called out previously. but, it would be a no-op due to the other task.